packages/utils/telemetry-utils/src/config.ts

```diff
@@ -0,0 +1,335 @@
+        return this.getConfig(name, "number") ?? defaultValue;
+    }
+
+    getString(name: string, defaultValue?: string): string | undefined {
+        return this.getConfig(name, "string") ?? defaultValue;
+    }
+
+    getBooleanArray(name: string, defaultValue?: boolean[]): boolean[] | undefined {
+        return this.getConfig(name, "boolean[]") ?? defaultValue;
+    }
+
+    getNumberArray(name: string, defaultValue?: number[]): number[] | undefined {
+        return this.getConfig(name, "number[]") ?? defaultValue;
+    }
+
+    getStringArray(name: string, defaultValue?: string[]): string[] | undefined {
+        return this.getConfig(name, "string[]") ?? defaultValue;
+    }
+
+    getRawConfig(name: string): ConfigTypes {
+        return this.getCacheEntry(name)?.raw;
+    }
+
+    private getConfig<T extends ConfigTypeStrings>(
+        name: string, converter: T,
+    ): ConfigTypeStringToType[T] | undefined {
+        const cacheValue = this.getCacheEntry(name);
+
+        if (cacheValue?.raw === undefined) {
+            ConfigProvider.logger.sendTelemetryEvent({ eventName: "ConfigValueNotSet", name: cacheValue?.name });
+            return undefined;
+        }
+
+        if (converter === "string" && cacheValue[converter] === undefined) {
+            return JSON.stringify(cacheValue.raw) as ConfigTypeStringToType[T];
+        }
+
+        if (converter in cacheValue) {
+            return cacheValue[converter] as ConfigTypeStringToType[T];
+        }
+
+        ConfigProvider.logger.sendErrorEvent({
+            eventName: "ConfigValueNotConvertible",
+            name: cacheValue?.name,
+            value: JSON.stringify(cacheValue),
+            converter,
+        });
+
+        return undefined;
+    }
+
+    private getCacheEntry(name: string): ConfigCacheEntry | undefined {
+        const namespacedName = this.namespace ? `${this.namespace}.${name}` : name;
+        if (!this.configCache.has(namespacedName)) {
+            for (const provider of this.orderedBaseProviders) {
+                const parsed = stronglyTypedParse(provider?.getRawConfig(namespacedName));
+                if (parsed !== undefined) {
+                    const entry: ConfigCacheEntry = {
+                        raw: parsed.value,
+                        name,
+                        [parsed.type]: parsed.value,
+                    };
+                    this.configCache.set(namespacedName, entry);
+                    return entry;
+                }
+            }
+            // configs are immutable, if the first lookup returned no results, all lookups should
+            this.configCache.set(namespacedName, { raw: undefined, name: namespacedName });
+        }
+        return this.configCache.get(namespacedName);
+    }
+}
+
+/**
+ * A type containing both a telemetry logger and a configuration provider
+ */
+export interface MonitoringContext<T extends ITelemetryBaseLogger = ITelemetryLogger> {
```

Comment:
open to naming suggestions. Ideas i had were: ServicingContext, MeasureingContext